app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -11,6 +11,10 @@ class OmniauthCallbacksController < Devise::OmniauthCallbacksController
 
     def handle(provider)
       auth_hash = request.env['omniauth.auth']
+      if auth_hash.info.email.nil?
```

Comment:
what about testing this as well? :thinking:

Reply:
@Ana06 If have written tests for this.